spring-cloud-starter-task-composedtaskrunner/src/main/java/org/springframework/cloud/task/app/composedtaskrunner/ComposedTaskRunnerConfiguration.java

```diff
@@ -0,0 +1,124 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.dataflow.rest.client.TaskOperations;
+import org.springframework.cloud.task.app.composedtaskrunner.properties.ComposedTaskProperties;
+import org.springframework.cloud.task.app.composedtaskrunner.properties.PropertyUtility;
+import org.springframework.cloud.task.configuration.EnableTask;
+import org.springframework.cloud.task.repository.TaskExecution;
+import org.springframework.cloud.task.repository.TaskExplorer;
+import org.springframework.cloud.task.repository.TaskRepository;
+import org.springframework.cloud.task.repository.support.SimpleTaskExplorer;
+import org.springframework.cloud.task.repository.support.SimpleTaskRepository;
+import org.springframework.cloud.task.repository.support.TaskExecutionDaoFactoryBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Configures the Job that will execute the Composed Task Execution.
+ *
+ * @author Glenn Renfro
+ */
+@Configuration
+@EnableBatchProcessing
+@EnableTask
+@EnableConfigurationProperties(ComposedTaskProperties.class)
+public class ComposedTaskRunnerConfiguration {
+
+	@Autowired
+	private JobBuilderFactory jobs;
+
+	@Autowired
+	private StepBuilderFactory steps;
+
+	@Autowired
+	private DataSource dataSource;
+
+	@Autowired
+	private ComposedTaskProperties properties;
+
+	@Autowired
+	private TaskOperations taskOperations;
+
+	@Bean
+	public Job job() throws Exception {
+		//TODO add parser generation of batch flow here
+		FlowBuilder<Flow> flowBuilder = new FlowBuilder<Flow>(UUID.randomUUID().toString())
+				.start(getStep(properties.getGraph(),
+						PropertyUtility.getPropertiesForTask(properties.getGraph(),
+								this.properties), null));//for now assumes single step until we get parser to build out the flow.
+		return this.jobs.get(
+				properties.getJobNamePrefix() + UUID.randomUUID().toString())
+				.start(flowBuilder.end()).end().build();
+	}
+
+	@Bean
+	public TaskRepository taskRepository() {
+		return new SimpleTaskRepository(new TaskExecutionDaoFactoryBean(dataSource));
+	}
+
+	@Bean
+	public TaskExplorer taskExplorer() {
+		return new SimpleTaskExplorer(new TaskExecutionDaoFactoryBean(dataSource));
+	}
+
+	private StepExecutionListener composedTaskStepExecutionListener() {
+		return new ComposedTaskStepExecutionListener(taskExplorer(), properties);
+	}
+
+	/**
+	 * Creates the step that will launch the TaskLauncherTasklet.
+	 * @param taskName - The name of the task definition to launch.
+	 * @param properties - the properties required for the task to be launched.
+	 * @param arguments - The command line arguments to be passed to the task.
+	 * @return The step that will be added to the job.
+	 */
+	private Step getStep(String taskName, Map<String, String> properties,
```

Comment:
These steps really should be registered with the context.  Right now, unless I'm missing something, the `InitailizingBean#afterPropertiesSet` on `AbstractStep` isn't being called with it configured this way.